### PR TITLE
FAQ: add MPI best practices

### PR DESCRIPTION
**Pull request by j-ogas**
Addresses: #379.

Dependent on #1449.

### CODE REVIEW

**Review by reidpr — APPROVED** [review]
Jordan, you asked if the tone is good. This is fantastic, more please!

**Review by reidpr — CHANGES_REQUESTED** [review]
Good work, nice and concise.

See my tidy and LMK if you like it.

File: doc/best_practices.rst [hunk 1]
```diff
@@ -303,5 +307,147 @@ building, and then run using a separate container invoked from a different
...
+
+Recommendations TL;DR
+---------------------
+
+Generally, we recommend building a flexible MPI container using:
+
+   a. **libfabric** to flexibly manage process communication over a diverse
+      set of network fabrics;
+
+   b. a parallel **process management interface** (PMI) compatible with the
```
Comment (j-ogas) [inline doc/best_practices.rst:370]:
```suggestion
   b. a parallel **process management interface** (PMI), e.g., pmix, pmi2,
      flux-pmi, etc., compatible with the host workload manager; and
```

File: doc/best_practices.rst [hunk 2]
```diff
@@ -303,5 +307,147 @@ building, and then run using a separate container invoked from a different
...
+In Charliecloud, both of these injection operations are currently done with
+:code:`ch-fromhost`, though see `issue #1861
+<https://github.com/hpc/charliecloud/issues/1861>`_.
+
+Choose a compatible PMI
+-----------------------
+
+Unprivileged processes, including unprivileged containerized processes, are
+unable to independently launch containerized processes on different nodes,
+aside from using SSH, which isn’t scalable. We must either (1)_rely on a host
```
Comment (j-ogas) [inline doc/best_practices.rst:430]:
```suggestion
aside from using SSH, which isn’t scalable. We must either (1) rely on a host
```

File: doc/best_practices.rst [hunk 3]
```diff
@@ -303,5 +307,147 @@ building, and then run using a separate container invoked from a different
...
+supported parallel process management interface (PMI), or (2)_achieve
```
Comment (j-ogas) [inline doc/best_practices.rst:431]:
```suggestion
supported parallel process management interface (PMI), or (2) achieve
```

File: doc/best_practices.rst [hunk 4]
```diff
@@ -303,5 +307,147 @@ building, and then run using a separate container invoked from a different
...
+
+The above compiles libfabric with several “built-in” providers, i.e.
+:code:`psm3` (on x86-64), :code:`rxm`, :code:`shm`, :code:`tcp`, and
+:code:`verbs`, which enables MPI applications to run efficiently over most
+verb devices using TCP, IB, OPA, and RoCE protocols.
+
+Two key advantages of using libfabric are: (1) the container’s libfabric can
+make use of “external” i.e. dynamic-shared-object (DSO) providers, and
+(2) Cray’s Slingshot provider (CXI) can be used by replacing the container
+image’s :code:`libfabric.so` with the Cray host’s.
```
Comment (j-ogas) [inline doc/best_practices.rst:409]:
```suggestion
Two key advantages of using libfabric are: (1) the container’s libfabric can
make use of “external”, i.e. dynamic-shared-object (DSO) providers, and
(2) libfabric replacement is much simpler than MPI replacement and preserves
the original container MPI, i.e., no host/container MPI ABI compatibility
issues; rather, we manage the more flexible libfabirc ABI compatibility. 
```

File: doc/best_practices.rst (hunk 2 above)
Comment (reidpr) [reply doc/best_practices.rst:430]: Thanks, this was supposed to be a non-breaking space, which shows up in my editor as a blue underscore.

**Review by reidpr — APPROVED** [review]

**Pull request merged**